lib/uefi/runtime_service_provider.cpp

```diff
@@ -61,6 +83,25 @@ EFI_STATUS SetVirtualAddressMap(size_t MemoryMapSize, size_t DescriptorSize,
 
 EFI_STATUS SetVariable(char16_t *VariableName, EfiGuid *VendorGuid,
                        uint32_t Attributes, size_t DataSize, void *Data) {
+  if (!VariableName || VariableName[0] == 0) {
+    return INVALID_PARAMETER;
+  }
+
+  /* Only allow setting non-volatile variables */
+  if ((Attributes & (~(EFI_VARIABLE_BOOTSERVICE_ACCESS | EFI_VARIABLE_RUNTIME_ACCESS))) == 0) {
```

Comment:
Check if volatile flag is set, then print a message saying only non-volatile variable is supported, and return UNSUPPORTED in volatile branch.